ground/src/main/java/com/google/android/ground/ui/map/gms/features/MapsItemManager.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.ui.map.gms.features
+
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Marker
+import com.google.android.gms.maps.model.Polygon as MapsPolygon
+import com.google.android.gms.maps.model.Polyline
+import com.google.android.ground.model.geometry.LineString
+import com.google.android.ground.model.geometry.MultiPolygon
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.geometry.Polygon
+import com.google.android.ground.ui.map.Feature
+import com.google.maps.android.PolyUtil.*
+
+/** Manages [Feature]s displayed on the map as Maps SDK items (marker, polyline, etc). */
+class MapsItemManager(
+  private val map: GoogleMap,
+  private val pointRenderer: PointRenderer,
+  private val polygonRenderer: PolygonRenderer,
+  private val lineStringRenderer: LineStringRenderer
+) {
+  private val itemsByTag = mutableMapOf<Feature.Tag, List<Any>>()
```

Comment:
Nit: kind of a shame to use `Any` and abandon all type specificity here imo -- can we target a supertype instead/add one? Feel free to ignore, as this is a bit pedantic (but it might catch a future bug!).

Feels like these types are amenable to a sealed class, which would also give use exhaustive compile-time `when` checking

Reply:
Unfortunately Maps doesn't offer a common supertype. I tried several variations of generic and refactorings, but because of the lack of a common supertype they all involved checking types at runtime someone. This was the simplest impl I could find.

Reply:
Dang! lgtm then, thanks for looking into it!